tripy/nvtripy/frontend/ops/plugin_qdp.py

```diff
@@ -0,0 +1,103 @@
+
+
+# TODO (pranavm): Add link to custom layers guide once published
+@export.public_api(document_under="operations/functions")
+def plugin(
+    op: str, inputs: Sequence["nvtripy.Tensor"], **kwargs: Dict[str, Any]
+) -> Union["nvtripy.Tensor", List["nvtripy.Tensor"]]:
+    """
+    Calls a TensorRT quickly deployable plugin (QDP).
+
+    Args:
+        op: The ID of plugin to call, in the form ``"<namespace>::<name>"``.
+        inputs: The inputs to the plugin.
+        **kwargs: Additional arguments to pass to the plugin as attributes.
+            Supported attribute types are
+            `documented here <https://docs.nvidia.com/deeplearning/tensorrt/latest/_static/python-api/infer/tensorrt.plugin/trt_plugin_register.html#tensorrt.plugin.register>`_.
+
+    Returns:
+        The output(s) of the plugin either as a single tensor if there is only one output,
+        or a list of tensors otherwise.
+    """
+    import tensorrt.plugin as trtp
+
+    if "::" not in op:
+        raise_error(f"Invalid plugin ID: '{op}'.", details=["Expected an ID of the form '<namespace>::<name>'."])
+    namespace, name = op.split("::")
+
+    try:
+        trtp_op = getattr(getattr(trtp.op, namespace), name)
+    except AttributeError:
+        available_namespaces = [
+            ns for ns in (getattr(trtp.op, attr) for attr in dir(trtp.op)) if isinstance(ns, trtp._lib._PluginNamespace)
+        ]
+
+        def get_plugins_in_namespace(ns):
+            return [attr for attr in dir(ns) if isinstance(getattr(ns, attr), trtp._lib.PluginDef)]
+
+        raise_error(
+            f"Plugin '{op}' not found.",
+            details=[
+                f"Note: Available namespaces and plugins are:\n"
+                + "\n".join(
+                    [
+                        f"- {ns._namespace}::{{ {', '.join(get_plugins_in_namespace(ns))} }}"
+                        for ns in available_namespaces
+                    ]
+                )
+            ],
+        )
+
+    attrs = {}
+    for key, value in kwargs.items():
+        if key not in trtp_op.input_attrs:
+            raise_error(
+                f"Unexpected attribute: '{key}'.",
+                details=[f"Available attributes are: {list(trtp_op.input_attrs.keys())}."],
+            )
+
+        attr_annotation = trtp_op.input_attrs[key]
+
+        try:
+            kwargs[key] = value.tolist()
+        except AttributeError:
+            if attr_annotation is not type(value):
+                raise_error(
+                    f"Unexpected attribute type: '{type_str_from_arg(value)}'.",
+                    details=[f"For attribute: '{key}', expected type: '{str_from_type_annotation(attr_annotation)}'."],
+                )
+
+        attrs[key] = value
+
+    input_descs = [None] * len(inputs)
+    for i in range(len(inputs)):
+        input_descs[i] = trtp._tensor.TensorDesc()
+        input_descs[i].dtype = inputs[i].dtype
```

Comment:
@samurdhikaru I just realized that this logic may not work. I assume QDPs are supposed to work with TRT dtypes (not Tripy ones)? In that case I think we'd need to convert accordingly?